rustler_codegen/src/ex_struct.rs

```diff
@@ -18,11 +18,22 @@ pub fn transcoder_decorator(ast: &syn::DeriveInput) -> TokenStream {
     // Unwrap is ok here, as we already determined that struct_fields is not None
     let field_atoms = ctx.field_atoms().unwrap();
 
-    let atom_defs = quote! {
-        rustler::atoms! {
-            atom_struct = "__struct__",
-            atom_module = #elixir_module,
```

Comment:
Maybe the exception atom could be quoted as well and inserted here. Then we don't need to duplicate `rustler::atoms!`. Something along these lines (untested!):

```rust
exception_quote =
if add_exception {
  quote!{atom_exception = "__exception__"}
} else {
  quote!{}
}
//...

rustler::atoms! {
  //...
  #exception_atom,
}
```

Reply:
Ah, that's actually quite similar to the encoding case (`exception_field`).